src/test/java/com/devonfw/sample/archunit/ThirdPartyRulesE1TransactionalTest.java

```diff
@@ -0,0 +1,57 @@
+package com.devonfw.sample.archunit;
+
+import com.tngtech.archunit.core.domain.Dependency;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class ThirdPartyRulesE1TransactionalTest {
+
+    private static boolean apiScopedClassIsUsingTransactional(JavaClass item, String targetPackageFullName) {
+        if(item.getFullName().contains("api") && targetPackageFullName.equals("javax.transaction.Transactional")) {
+                return true;
+        }
+        return false;
+    }
+
+    private static boolean isUsingSpringframeworkTransactionalAnnotation(String targetPackageFullName) {
+        if(targetPackageFullName.equals("org.springframework.transaction.annotation.Transactional")) {
+                return true;
+        }
+        return false;
+    }
+
+    static ArchCondition<JavaClass> miuse_springframework_transactional_annotation = new ArchCondition<JavaClass> ("misuse @Transactional (Rule-E1)") {
+        @Override
+        public void check(JavaClass item, ConditionEvents events) {
+                for(Dependency access: item.getDirectDependenciesFromSelf()) {
+                        String targetPackageFullName = access.getTargetClass().getFullName();
```

Comment:
The `Package` in the variable name is IMHO misleading as it is a qualified name of a type / class and not just a package name.

Reply:
Indeed, that's a poor name choice. I changed the variable name `targetPackageFullName` to `targetFullName`. 
In addition, I changed the first parameter name of the _check_ method from `item` to `sourceClass,` which leads to better code readability and understanding.
Also, I changed the Dependency name in Row:35 from `access` to `dependency,` since access might also be misleading. One might think there's actual access towards a forbidden class, whereas even declaring a field of an undesired class type will resolve into an error message.

Reply:
OK. Thanks. Great.